src/candle_components/candle_datetime.rs

```diff
@@ -0,0 +1,33 @@
+#![cfg(feature = "chrono")]
+use chrono::{DateTime, Utc};
+
+#[derive(Debug, Clone)]
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+pub struct CandleDateTime {
+    init: bool,
+    value: DateTime<Utc>,
+}
+
+impl CandleComponent<DateTime<Utc>> for CandleDateTime {
+    /// Returns the open price of the candle
+    #[inline(always)]
+    fn value(&self) -> DateTime<Utc> {
+        self.value
+    }
+    /// This makes sure the next time "update" is called, the new open value is set
+    #[inline(always)]
+    fn reset(&mut self) {
+        self.init = true;
+    }
+}
+
+impl<T: TakerTrade> CandleComponentUpdate<T> for CandleDateTime {
+    /// Only update the open price if this module is in init mode
+    #[inline(always)]
+    fn update(&mut self, trade: &T) {
+        if self.init {
+            self.value = Utc.timestamp_millis(trade.timestamp(), 0);
```

Comment:
The assumption that timestamps are always in milliseconds does not generally hold. There must be a clean way of distinguishing between the timestamp resolutions. The crude way would be to just append `Millis` to the name of the `struct`. Maybe you can figure out a better way, I'm happy either way, as long as the assumption is documented (preferably on the `struct`).

Reply:
Hmmm, which `struct` do you mean?

We do document it on the `TakerTrade` trait here:

```
/// Trait to enable third party types to be passed into aggregators.
pub trait TakerTrade {
    /// The timestamp of a trade,
    /// For the built in time rule, this is expected to be in milliseconds
    fn timestamp(&self) -> i64;
```

Reply:
Good catch. I think the comment that its expected in milliseconds can be remove and we can leave it up to the user. I've been using nanoseconds for a while without any problems, because the `TimeRule` has the option to work with multiple timestamp resolutions. Not sure if the same can be applied to that `CandleComponent`, because it relies on the `Default` constructor. Therefore appending `Millis` to the `OpenDateTime` struct would clarify that it expects milisecond timestamps. Its not the prettiest workaround but would be a quick fix. So if a user would like `Nanos` he would have to clone that file and modify the timestamp parsing accordingly.

Reply:
Sounds good, I just pushed what I was thinking.  It's just a bolt on really, a more involved approach would have been to have `timestamp` return an enum with the value embedded but that didn't seem worth it.  This keeps things simple and the `OpenDateTime` struct is the only one using this new method.

I will be traveling starting tomorrow and would like to add a simple test prior to merging, I just don't have the time atm.  I will keep you posted once that is done.  I don't know if I can add a test without making the `chrono` feature a default feature so maybe the test will be an integration or example crate, any thoughts there?

Reply:
Sounds good. I think if the whole module is feature gated behind `chrono` you should be able to embed a `tests` module and it should work. Otherwise just make `chrono` a normal dependency, that's fine too.

Reply:
I was able to get testing to work by passing `--all-features` as a flag to `cargo test`.

Just added a couple simple tests.